test/test-test.js

```diff
@@ -218,7 +219,24 @@ buster.testCase("sinon-test", {
             callback();
         }).call({}, "arg1", {}, done);
     },
-
+    "promises": {
+        "cleanup on resolution": function (done) {
+            var obj = {foo: function () {}};
+            var promise = instance(function () {
+                var spy = this.spy(obj, "foo");
+                return Promise.delay(1).then(function () {
+                    obj.foo();
+                }).then(function () {
+                    assert(spy.calledOnce);
+                });
+            }).call({});
+            return promise.then(function () {
```

Comment:
You don't need to wrap the functions in a closure. Just omit the enclosing function completely:
`return promise.then(done).catch(done);`